src/probnum/filtsmooth/gaussfiltsmooth/kalman.py

```diff
@@ -1,128 +1,213 @@
-    GaussFiltSmooth,
-    linear_discrete_update,
-)
-from probnum.filtsmooth.statespace import DiscreteGaussianLinearModel, LinearSDEModel
+from probnum._randomvariablelist import _RandomVariableList
+from probnum.filtsmooth.bayesfiltsmooth import BayesFiltSmooth
+from probnum.filtsmooth.gaussfiltsmooth.kalmanposterior import KalmanPosterior
 from probnum.random_variables import Normal
 
 
-class Kalman(GaussFiltSmooth):
+class Kalman(BayesFiltSmooth):
     """
-    Kalman filtering and smoothing for continuous-discrete and
-    discrete-discrete state space models.
+    Gaussian filtering and smoothing, i.e. Kalman-like filters and smoothers.
     """
 
-    def __new__(cls, dynamod, measmod, initrv, **kwargs):
-        """
-        Factory method for Kalman filtering and smoothing.
+    def __init__(self, dynamic_model, measurement_model, initrv):
+        """Check that the initial distribution is Gaussian."""
+        if not issubclass(type(initrv), Normal):
+            raise ValueError(
+                "Gaussian filters/smoothers need initial "
+                "random variables with Normal distribution."
+            )
+        super().__init__(dynamic_model, measurement_model, initrv)
 
-        Depending on whether the dynamic model is continuous or
-        discrete, either a continuous-discrete Kalman object or a
-        discrete-discrete Kalman object is created.
+    def filtsmooth(self, dataset, times, **kwargs):
         """
-        if cls is Kalman:
-            if is_cont_disc(dynamod, measmod):
-                return _ContDiscKalman(dynamod, measmod, initrv, **kwargs)
-            if is_disc_disc(dynamod, measmod):
-                return _DiscDiscKalman(dynamod, measmod, initrv)
-            else:
-                errmsg = (
-                    "Cannot instantiate Kalman object with given "
-                    "dynamic model and measurement model."
-                )
-                raise ValueError(errmsg)
-        else:
-            return super().__new__(cls)
-
-
-class _ContDiscKalman(Kalman):
-    """
-    Provides predict() and update() methods for Kalman filtering and
-    smoothing on continuous-discrete state space models.
-    """
+        Apply Gaussian filtering and smoothing to a data set.
+
+        Parameters
+        ----------
+        dataset : array_like, shape (N, M)
+            Data set that is filtered.
+        times : array_like, shape (N,)
+            Temporal locations of the data points.
+        kwargs : ???
+
+        Returns
+        -------
+        KalmanPosterior
+            Posterior distribution of the smoothed output
+        """
+        dataset, times = np.asarray(dataset), np.asarray(times)
+        filter_posterior = self.filter(dataset, times, **kwargs)
+        smooth_posterior = self.smooth(filter_posterior, **kwargs)
+        return smooth_posterior
 
-    def __init__(self, dynamod, measmod, initrv, **kwargs):
+    def filter(self, dataset, times, **kwargs):
```

Comment:
they are there for interface reasons. The superclass has them, so I put them here for peer pressure reasons... Thoughts?

Reply:
I don't know about the best practices regarding function signatures of classes and subclasses. I guess they should be the same?

In any case, I'd be in favor of removing **kwargs wherever we don't explicitly need them. But I also understand that we might want to do that separately and not in this PR (although that seems like one of those issues that we never explicitly tackle), so it's also fine for me if you want to keep them in. If we postpone, it would be nice to open an issue for it.

Reply:
lets do this in a separate PR. I will create an issue.